### DISCUSSION TITLE
Daterange implementations in postgresql..

### DISCUSSION THREAD

**Discussion by swnesbv** [diesel-rs/diesel#4088] [Q&A]
There is a proven sql implementation of postgresql(daterange)..
`
"SELECT * FROM provision_d WHERE st_date <= $1 AND en_date >= $2 AND NOT daterange($1, $2, '[]') @> ANY(dates) OR dates IS NULL"
`
I find it difficult to implement the conversion to "diesel"..
In fact, only primitives work - (<, >, IS NULL)..
"Available on crate feature postgres_backend only.": - And where is it..?
`

    let custom_query = "SELECT * FROM provision_d WHERE st_date <= $1 AND en_date >= $2 AND NOT daterange($1, $2, '[]') @> ANY(dates) OR dates IS NULL";
    
    use diesel::sql_query;
    use diesel::sql_types::Date;
    let pr_list:Vec<SqlPrD> = sql_query(custom_query)
        .bind::<Date, NaiveDate>(k.start)
        .bind::<Date, NaiveDate>(k.end)
        .load::<SqlPrD>(&mut conn)
        .await
        .unwrap();

`

**Comment by weiznich** [comment]
It's unclear what you want to know. You might want to add additional information to your question:

* What diesel version do you use
* Which database backends do you use 
* Which operating system do you use
* Which rust version do you use
* Which crate features are enabled 
* Are there any errors? If yes which?


> Available on crate feature postgres_backend only.

That means that using this specific type/function/.. requires that you enable the `postgres_backend` feature for that crate.

**Reply by swnesbv** [reply]
Here is the necessary functionality..
https://www.postgresql.org/search/?u=%2Fdocs%2F16%2F&q=daterange
Here is the solution for (golang, sql)..
https://github.com/swnesbv/go-example/blob/main/booking/repository.go
We need a solution (rust, diesel)..
https://github.com/swnesbv/axum-example/blob/main/src/booking/handlers.rs

**Reply by weiznich** [reply]
To repeat it one last time: It's unclear what you exactly want to know. You really need to provide at least the information we asked for above otherwise it's not possible to help you.

**Reply by swnesbv** [reply]
Guys, you would go outside and ask there: what date is today and what planet is this..

**Reply by weiznich** [reply]
I just want to state that this behaviour is not welcome here. You are obviously free to dismiss these questions, just don't expect that anyone will then spend time on understanding and resolving your problem. In that case you are on your own there.
